File: MyApplication/app/src/main/java/vinh/tan/tam/com/myapplication/MainActivity.java
package vinh.tan.tam.com.myapplication;

import android.os.Bundle;
import android.util.Log;

import java.util.List;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    RVAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = (RecyclerView) findViewById(R.id.recycler);
        adapter = new RVAdapter();
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://mapi-ng.rdc.moveaws.com/")
                .build();

        RetrofitService service = retrofit.create(RetrofitService.class);
        Call<List<Repo>> repos = service.listRepos();
        repos.enqueue(new Callback<List<Repo>>() {
            @Override
            public void onResponse(Call<List<Repo>> call, Response<List<Repo>> response) {
                Log.d("response_data", response.body().toString());

            }

            @Override
            public void onFailure(Call<List<Repo>> call, Throwable t) {
                Log.d("response_data", t.toString());

            }
        });
    }
}

File: RetrofitWithKotlin/Calculator/app/src/main/java/vinh/tan/tam/com/calculator/MainActivity.kt
package vinh.tan.tam.com.calculator

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var first: Int? = null
    var oper: Char? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnAdd.setOnClickListener({
            if (edtInput.text.length != 0) {
                first = first?.plus(edtInput.text.toString().toInt())
                    ?: edtInput.text.toString().toInt()
                edtInput.setText("")
                oper = '+'
            }
        })

        btnSub.setOnClickListener({
            if (edtInput.text.length != 0) {
                first = first?.minus(edtInput.text.toString().toInt())
                    ?: edtInput.text.toString().toInt()
                edtInput.setText("")
                oper = '-'
            }
        })
        btnMul.setOnClickListener({
            if (edtInput.text.length != 0) {
                first = first?.times(edtInput.text.toString().toInt())
                    ?: edtInput.text.toString().toInt()
                edtInput.setText("")
                oper = '*'
            }
        })


        btnDiv.setOnClickListener({
            if (edtInput.text.length != 0) {
                first =
                    first?.div(edtInput.text.toString().toInt()) ?: edtInput.text.toString().toInt()
                edtInput.setText("")
                oper = '/'
            }
        })

        btnClear.setOnClickListener({
            if (edtInput.text.length != 0) {
                edtInput.setText("")
            }
            first = null
        })

        btnEqual.setOnClickListener({
            if (first != null) {
                Log.e("calculatorlog", first.toString())
                when (oper) {
                    '+' -> {
                        if (edtInput.text.length != 0) {
                            first = first?.plus(edtInput.text.toString().toInt())
                                ?: edtInput.text.toString().toInt()
                        }
                    }
                    '-' -> {
                        if (edtInput.text.length != 0) {
                            first = first?.minus(edtInput.text.toString().toInt())
                                ?: edtInput.text.toString().toInt()
                        }
                    }
                    '*' -> {
                        if (edtInput.text.length != 0) {
                            first = first?.times(edtInput.text.toString().toInt())
                                ?: edtInput.text.toString().toInt()
                        }
                    }
                    '/' -> {
                        if (edtInput.text.length != 0) {
                            first = first?.div(edtInput.text.toString().toInt())
                                ?: edtInput.text.toString().toInt()
                        }
                    }
                    else -> {

                    }
                }
                edtInput.setText("$first")
                first = null
            }
        })
    }

}
